src/main/scala/com/microsoft/ml/spark/explainers/SharedParams.scala

```diff
@@ -0,0 +1,193 @@
+}
+
+trait HasTokensCol extends Params {
+  final val tokensCol = new Param[String](
+    this,
+    "tokensCol",
+    "The column holding the tokens"
+  )
+
+  final def getTokensCol: String = $(tokensCol)
+
+  final def setTokensCol(v: String): this.type = this.set(tokensCol, v)
+}
+
+trait HasSuperpixelCol extends Params {
+  final val superpixelCol = new Param[String](
+    this,
+    "superpixelCol",
+    "The column holding the superpixel decompositions"
+  )
+
+  final def getSuperpixelCol: String = $(superpixelCol)
+
+  final def setSuperpixelCol(v: String): this.type = set(superpixelCol, v)
+}
+
+trait HasSamplingFraction extends Params {
+  final val samplingFraction = new DoubleParam(
+    this,
+    "samplingFraction",
+    "The fraction of superpixels (for image) or tokens (for text) to keep on",
+    ParamValidators.inRange(0, 1)
+  )
+
+  final def getSamplingFraction: Double = $(samplingFraction)
+
+  final def setSamplingFraction(d: Double): this.type = set(samplingFraction, d)
+}
+
+trait HasBackgroundData extends Params {
+  final val backgroundData: DataFrameParam = new DataFrameParam(
+    this,
+    "backgroundData",
+    "A dataframe containing background data"
+  )
+
+  final def getBackgroundData: DataFrame = $(backgroundData)
+
+  final def setBackgroundData(value: DataFrame): this.type = set(backgroundData, value)
+}
+
+trait HasExplainTarget extends Params {
+  final val targetCol: Param[String] = new Param[String](
+    this,
+    "targetCol",
+    "The column name of the prediction target to explain (i.e. the response variable). " +
+      "This is usually set to \"prediction\" for regression models and " +
+      "\"probability\" for probabilistic classification models. Default value: probability")
+  setDefault(targetCol, "probability")
+
+  final def getTargetCol: String = $(targetCol)
+  final def setTargetCol(value: String): this.type = this.set(targetCol, value)
+
+  final val targetClasses: IntArrayParam = new IntArrayParam(
+    this,
+    "targetClasses",
+    "The indices of the classes for multinomial classification models. Default: 0." +
+      "For regression models this parameter is ignored."
+  )
+
+  final def getTargetClasses: Array[Int] = $(targetClasses)
+  final def setTargetClasses(values: Array[Int]): this.type = this.set(targetClasses, values)
+
+  final val targetClassesCol: Param[String] = new Param[String](
+    this,
+    "targetClassesCol",
+    "The name of the column that specifies the indices of the classes for multinomial classification models."
+  )
+
+  final def getTargetClassesCol: String = $(targetClassesCol)
+  final def setTargetClassesCol(value: String): this.type = this.set(targetClassesCol, value)
+
+  private def slice[T](values: Int => T, indices: Seq[Int])(num: Numeric[_]): SV = {
+    val n = num.asInstanceOf[Numeric[T]]
+    SVS.dense(indices.map(values.apply).map(n.toDouble).toArray)
+  }
+
+  private val dataTypeToNumericMap: Map[NumericType, Numeric[_]] = Map(
```

Comment:
interesting utility, I wonder if we should put other core utilities like this in a more core-place in the repo (instead of under explainers), what do you think?

Reply:
done. Extracted to `com.microsoft.ml.spark.core.utils.SlicerFunctions` and added unit tests.